src/gen/conditions.rs

```diff
@@ -253,94 +249,103 @@ pub fn parse_args(
-            // both negative values and values exceeding the max are invalid
-            let s = match sanitize_uint(a, first(a, c)?, 8, code) {
-                Err(ValidationErr(n, ErrorCode::NegativeAmount)) => Err(ValidationErr(n, code)),
-                Err(ValidationErr(n, ErrorCode::AmountExceedsMaximum)) => {
-                    Err(ValidationErr(n, code))
-                }
-                Err(r) => Err(r),
-                Ok(r) => Ok(u64_from_bytes(r)),
-            }?;
-
-            Ok(Condition::AssertMyBirthSeconds(s))
+            match sanitize_uint(a, node, 8, code)? {
+                SanitizedUint::PositiveOverflow => Err(ValidationErr(node, code)),
+                SanitizedUint::NegativeOverflow => Err(ValidationErr(node, code)),
+                SanitizedUint::Ok(r) => Ok(Condition::AssertMyBirthSeconds(r)),
+            }
         }
         ASSERT_MY_BIRTH_HEIGHT => {
             maybe_check_args_terminator(a, c, flags)?;
+            let node = first(a, c)?;
             let code = ErrorCode::AssertMyBirthHeightFailed;
-            // both negative values and values exceeding the max are invalid
-            let h = match sanitize_uint(a, first(a, c)?, 4, code) {
-                Err(ValidationErr(n, ErrorCode::NegativeAmount)) => Err(ValidationErr(n, code)),
-                Err(ValidationErr(n, ErrorCode::AmountExceedsMaximum)) => {
-                    Err(ValidationErr(n, code))
-                }
-                Err(r) => Err(r),
-                Ok(r) => Ok(u64_from_bytes(r) as u32),
-            }?;
-            Ok(Condition::AssertMyBirthHeight(h))
+            match sanitize_uint(a, node, 4, code)? {
+                SanitizedUint::PositiveOverflow => Err(ValidationErr(node, code)),
+                SanitizedUint::NegativeOverflow => Err(ValidationErr(node, code)),
+                SanitizedUint::Ok(r) => Ok(Condition::AssertMyBirthHeight(r as u32)),
+            }
         }
         ASSERT_SECONDS_RELATIVE => {
             maybe_check_args_terminator(a, c, flags)?;
-            match sanitize_uint(a, first(a, c)?, 8, ErrorCode::AssertSecondsRelative) {
-                // A negative argument for ASSERT_SECONDS_RELATIVE is always true
-                Err(ValidationErr(_, ErrorCode::NegativeAmount)) => Ok(Condition::Skip),
-                Err(ValidationErr(n, ErrorCode::AmountExceedsMaximum)) => {
-                    Err(ValidationErr(n, ErrorCode::AssertSecondsRelative))
-                }
-                Err(r) => Err(r),
-                Ok(r) => Ok(Condition::AssertSecondsRelative(u64_from_bytes(r))),
+            let node = first(a, c)?;
+            let code = ErrorCode::AssertSecondsRelative;
+            match sanitize_uint(a, node, 8, code)? {
+                SanitizedUint::PositiveOverflow => Err(ValidationErr(node, code)),
+                SanitizedUint::NegativeOverflow => Ok(Condition::Skip),
+                SanitizedUint::Ok(r) => Ok(Condition::AssertSecondsRelative(r)),
```

Comment:
In general, having `x` and `before x` next to each other helps highlighting the main difference in their arg parsing here (the positive and negative overflow negation) which would improve readability.

Reply:
I'm hesitant to move code in the same PR as I update it. but it's a good point that it would probably hightlight the differences better